plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakePerTransactionMetastoreCache.java

```diff
@@ -16,96 +16,89 @@
-        deltaLakeProperties.put("hive.s3.path-style-access", "true");
-        deltaLakeProperties.put("hive.metastore", "test"); // use test value so we do not get clash with default bindings)
-        deltaLakeProperties.put("delta.register-table-procedure.enabled", "true");
-        if (!enablePerTransactionHiveMetastoreCaching) {
-            // almost disable the cache; 0 is not allowed as config property value
-            deltaLakeProperties.put("delta.per-transaction-metastore-cache-maximum-size", "1");
-        }
-
-        queryRunner.createCatalog(DELTA_CATALOG, "delta_lake", deltaLakeProperties.buildOrThrow());
-
-        if (createdDeltaLake) {
-            List<TpchTable<? extends TpchEntity>> tpchTables = List.of(TpchTable.NATION, TpchTable.REGION);
-            tpchTables.forEach(table -> {
+        try {
+            FileHiveMetastore fileMetastore = createTestingFileHiveMetastore(queryRunner.getCoordinator().getBaseDataDir().resolve("file-metastore").toFile());
+            metastore = new CountingAccessHiveMetastore(fileMetastore);
+            queryRunner.installPlugin(new TestingDeltaLakePlugin(Optional.empty(), Optional.empty(), new CountingAccessMetastoreModule(metastore)));
+
+            ImmutableMap.Builder<String, String> deltaLakeProperties = ImmutableMap.builder();
+            deltaLakeProperties.put("hive.metastore", "test"); // use test value so we do not get clash with default bindings)
+            deltaLakeProperties.put("delta.register-table-procedure.enabled", "true");
+            if (!enablePerTransactionHiveMetastoreCaching) {
+                // almost disable the cache; 0 is not allowed as config property value
+                deltaLakeProperties.put("delta.per-transaction-metastore-cache-maximum-size", "1");
+            }
+
+            queryRunner.createCatalog(DELTA_CATALOG, "delta_lake", deltaLakeProperties.buildOrThrow());
+            queryRunner.execute("CREATE SCHEMA " + session.getSchema().orElseThrow());
+
+            for (TpchTable<? extends TpchEntity> table : List.of(TpchTable.NATION, TpchTable.REGION)) {
                 String tableName = table.getTableName();
-                hiveMinioDataLake.copyResources("io/trino/plugin/deltalake/testing/resources/databricks/" + tableName, tableName);
-                queryRunner.execute(format("CALL %1$s.system.register_table('%2$s', '%3$s', 's3://%4$s/%3$s')",
-                        DELTA_CATALOG,
-                        "default",
-                        tableName,
-                        BUCKET_NAME));
-            });
+                String resourcePath = "io/trino/plugin/deltalake/testing/resources/databricks/" + tableName + "/";
+                Path tableDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("%s-%s".formatted(tableName, randomNameSuffix()));
+
+                for (ClassPath.ResourceInfo resourceInfo : ClassPath.from(MinioClient.class.getClassLoader())
```

Comment:
`MinioClient` looks a little weird here now that you're not using Minio, is that just because it's in the testing module?
You could also CTAS the tables rather than loading the resources.

Reply:
> `MinioClient` looks a little weird here now that you're not using Minio, is that just because it's in the testing module?

good catch. left-over of inlining

> You could also CTAS the tables rather than loading the resources.

i didn't want to change test semantics (static input vs what current version would produce), but i am open to such change in the future